Contributing adjustments (#425)
cisagov/log4j-affected-db: PULL-EXAMPLE.md

# Pull Request Example #

Thank you for contributing to CISA's Log4j-affected-db GitHub Repository! Please
follow the steps listed below in order to add a product to the public
repository. **Note:** To assure the accuracy of data please only include
products that have official advisories or alerts that verify the product's
vulnerability status to  [CVE-2021-4104](https://nvd.nist.gov/vuln/detail/cve-2021-4104),
[CVE-2021-44228](https://nvd.nist.gov/vuln/detail/CVE-2021-44228),
[CVE-2021-45046](https://nvd.nist.gov/vuln/detail/CVE-2021-45046),
and/or [CVE-2021-45105](https://nvd.nist.gov/vuln/detail/CVE-2021-45105?s=09).

> **Step 1:** Go to the cisagov data .yml files and choose the appropriate file to
> include your updates (example link). The files are separated alphabetically
> based on the first letter of the vendor name. For example, 'CISA' would be
> located in [`cisagov_C.yml`](https://github.com/cisagov/data/cisagov_C.yml).

Reviewer: Here are some suggestions for this block:
* Add a link to the data directory (won't work right now, but will be correct after this PR is merged)
* Remove the "example link" text
* Correct the `cisagov_C.yml` URL (won't work right now, but will be correct after this PR is merged)
```suggestion
**Step 1:** Go to the [cisagov data .yml files](https://github.com/cisagov/log4j-affected-db/tree/develop/data/)
and choose the appropriate file to include your updates. 
The files are separated alphabetically based on the first letter of the
vendor name. For example, 'CISA' would be located in 
[`cisagov_C.yml`](https://github.com/cisagov/log4j-affected-db/blob/develop/data/cisagov_C.yml).
```

Reviewer: @Lcerkov - sorry about that trailing whitespace in my suggestion above.  Thanks for fixing it in https://github.com/cisagov/log4j-affected-db/pull/425/commits/38fe6123ee712b7f35d3d31e014c7b0e6431767b. 👍

follow the steps listed below in order to add a product to the public
repository. **Note:** To assure the accuracy of data please only include
products that have official advisories or alerts that verify the product's
vulnerability status to  [CVE-2021-4104](https://nvd.nist.gov/vuln/detail/cve-2021-4104),
[CVE-2021-44228](https://nvd.nist.gov/vuln/detail/CVE-2021-44228),
[CVE-2021-45046](https://nvd.nist.gov/vuln/detail/CVE-2021-45046),
and/or [CVE-2021-45105](https://nvd.nist.gov/vuln/detail/CVE-2021-45105?s=09).

**Step 1:** Go to the [cisagov data .yml files](https://github.com/cisagov/log4j-affected-db/tree/develop/data/)
and choose the appropriate file to include your updates.
The files are separated alphabetically based on the first letter of the
vendor name. For example, 'CISA' would be located in
[`cisagov_C.yml`](https://github.com/cisagov/log4j-affected-db/blob/develop/data/cisagov_C.yml).

**Step 2:** Click the file edit button.

**Step 3:** Add the blank template to the **.yml** file for new entries and fill
it out with the correct data.

Blank template:

```yml
  - vendor: ''
    product: ''

>     cves:

Reviewer 2: Recommend adding **CVE-2021-44832** to template per #404

      - ''
    last_updated: ''
```

**Step 4:** Verify the new entry was entered before and/or after the prior and
next entries.

- If you are adding it to the beginning of the file, ensure you are pasting
it after `software:` and before the next entry starting with `- vendor:`.
- If you are adding it to the end of the file, ensure the entry ends before
the file is closed out with `...`.

**Step 5:** When you add content to the file, remove the `''` or `[]` for fields
which are replaced with values (strings `''` or lists `[]` should be replaced as
shown in the example below, with list values on the following line(s) starting
with hyphen(s) `-`). The symbols should only remain used when fields remain
empty. For example, `fixed_versions: []` in our example below remains as is,
given there are no patched versions available for this entry.

**Note:** not all fields have to be updated.

Entry example:

```yml
  - vendor: Example Vendor
    product: Example Product
    cves:
      cve-2021-4104:
        investigated: false
        affected_versions: []
        fixed_versions: []
        unaffected_versions: []
      cve-2021-44228:
        investigated: true
        affected_versions:

>         - <=8.4.6
>         - <=8.5.3
>         - <=8.6.4

Reviewer 3: ```suggestion
          - <=8.4.6
          - <=8.5.3
          - <=8.6.4
```

        fixed_versions: []
        unaffected_versions: []
      cve-2021-45046:
        investigated: ''
        affected_versions: []
        fixed_versions: []
        unaffected_versions: []
      cve-2021-45105:
        investigated: ''
        affected_versions: []
        fixed_versions: []
        unaffected_versions: []
    vendor_links:
      - https://www.example.org/
    notes: Contains vulnerable code but not likely to get

>     unauthenticated user input to the log4j component.

Reviewer 3: ```suggestion
      unauthenticated user input to the log4j component.
```

    references:
      - ''
    last_updated: '2021-12-14T00:00:00'
```

**Step 6:** Validate that your data follows the appropriate format and proceed
with submitting the pull request.

> For any additional questions feel free to [submit an Issue request](https://github.com/cisagov/log4j-affected-db/issues).

Reviewer 3: ```suggestion
For any additional questions feel free to [submit an Issue](https://github.com/cisagov/log4j-affected-db/issues).
```

> Thank you for contributing to CISA's Log4j-affected-db GitHub Repository! Please follow the steps listed below in order to add a product to the public repository. **Note:** To verify the accuracy of data please only include products that have official advisories or alerts to verify the product's vulnerabilty status to  [CVE-2021-4104](https://nvd.nist.gov/vuln/detail/cve-2021-4104), [CVE-2021-44228](https://nvd.nist.gov/vuln/detail/CVE-2021-44228), [CVE-2021-45046](https://nvd.nist.gov/vuln/detail/CVE-2021-45046) and/or [CVE-2021-45105.](https://nvd.nist.gov/vuln/detail/CVE-2021-45105?s=09)

Reviewer 4: ```suggestion
Thank you for contributing to CISA's Log4j-affected-db GitHub Repository! Please follow the steps listed below in order to add a product to the public repository. **Note:** To assure the accuracy of data please only include products that have official advisories or alerts that verify the product's vulnerabilty status to  [CVE-2021-4104](https://nvd.nist.gov/vuln/detail/cve-2021-4104), [CVE-2021-44228](https://nvd.nist.gov/vuln/detail/CVE-2021-44228), [CVE-2021-45046](https://nvd.nist.gov/vuln/detail/CVE-2021-45046), and/or [CVE-2021-45105.](https://nvd.nist.gov/vuln/detail/CVE-2021-45105?s=09).
```

> **Step 1:** Go to the cisagov data .yml files for choosing the appropriate file to include your updates (example link). The files are separated based on the alphabetical first letter of the vendor name. For example, 'CISA' would be located in https://github.com/cisagov/data/cisagov_C.yml

Reviewer 4: ```suggestion
**Step 1:** Go to the cisagov data .yml files and choose the appropriate file to include your updates (example link). The files are separated alphabetically based on the first letter of the vendor name. For example, 'CISA' would be located in [`cisagov_C.yml`](https://github.com/cisagov/data/cisagov_C.yml).
```

**Step 2:** Click the file edit button.

**Step 3:** Add the blank template to the **.yml** file for new entries and fill it out with the correct data.  

Blank template:
```
- vendor: ''
  product: ''
  cves:
    cve-2021-4104:
      investigated: ''
      affected_versions: []
      fixed_versions: []
      unaffected_versions: []
    cve-2021-44228:
      investigated: ''
      affected_versions: []
      fixed_versions: []
      unaffected_versions: []
    cve-2021-45046:
      investigated: ''
      affected_versions: []
      fixed_versions: []
      unaffected_versions: []
    cve-2021-45105:
      investigated: ''
      affected_versions: []
      fixed_versions: []
      unaffected_versions: []
  vendor_links:
  - ''
  notes: ''
  references:
  - ''
  last_updated: ''
```
**Step 3:** Verify the new entry was entered before and/or after the prior and next entries.

>    - If you are adding it to the beginning of the file, ensure you are pasting it after “software:” and before the next entry starting with “- vendor:”.
>    -	If you are adding it to the end of the file, ensure the entry ends before the file is closed out with: “...”

Reviewer 4: ```suggestion
   - If you are adding it to the beginning of the file, ensure you are pasting it after `software:` and before the next entry starting with `- vendor:`.
   -	If you are adding it to the end of the file, ensure the entry ends before the file is closed out with `...`.
```

> **Step 4:** When you add content to the file, remove the '' or [] for fields which are replaced with values (strings '' or lists [] should be replaced as shown in the example below, with list values on the following line(s), starting with hyphen(s) “-”). The symbols should only remain used when fields remain empty. For example, fixed_versions: [] in our example below remains as is, given there are no patched versions available for this entry.

Reviewer 4: ```suggestion
**Step 4:** When you add content to the file, remove the `''` or `[]` for fields which are replaced with values (strings `''` or lists `[]` should be replaced as shown in the example below, with list values on the following line(s) starting with hyphen(s) `-`). The symbols should only remain used when fields remain empty. For example, `fixed_versions: []` in our example below remains as is, given there are no patched versions available for this entry.
```

**Note:** not all fields have to be updated.

Entry example:
```
- vendor: CISA
  product: Product
  cves:
    cve-2021-4104:
      investigated: false
      affected_versions: []
      fixed_versions: []
      unaffected_versions: []
    cve-2021-44228:
      investigated: true
      affected_versions:
      - <=8.4.6
      - <=8.5.3
      - <=8.6.4
      fixed_versions: []
      unaffected_versions: []
    cve-2021-45046:
      investigated: false
      affected_versions: []
      fixed_versions: []
      unaffected_versions: []
    cve-2021-45105:
      investigated: false
      affected_versions: []
      fixed_versions: []
      unaffected_versions: []
  vendor_links:

>   - 

Reviewer 4: I suggest putting a dummy value here instead of an empty list item (i.e., the existing line containing just a single hyphen character).  If you want to leave this field blank then you should set it to an empty list, as is done, e.g., for `unaffected_versions`.
```suggestion
  - https://example.org/
```

  notes: Contains vulnerable code but not likely to get
  unauthenticated user input to the log4j component.
  references:
  - ''
  last_updated: '2021-12-14T00:00:00'
```
**Step 5:** Validate that your data follows the appropriate format and proceed with submitting the pull request. 

> For any additional questions feel free to submit an Issue request [here](https://github.com/cisagov/log4j-affected-db/issues)

Reviewer 4: ```suggestion
For any additional questions feel free to [submit an Issue request](https://github.com/cisagov/log4j-affected-db/issues).
```
